atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/knowledge/AbortedTransactionSoftCache.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.transaction.knowledge;
+
+import com.palantir.atlasdb.autobatch.Autobatchers;
+import com.palantir.atlasdb.autobatch.BatchElement;
+import com.palantir.atlasdb.autobatch.DisruptorAutobatcher;
+import com.palantir.atlasdb.futures.AtlasFutures;
+import com.palantir.logsafe.Preconditions;
+import java.time.Duration;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicReference;
+
+public final class AbortedTransactionSoftCache implements AutoCloseable {
+    public enum TransactionSoftCacheStatus {
+        PENDING_LOAD_FROM_RELIABLE,
+        IS_ABORTED,
+        IS_NOT_ABORTED;
+    }
+
+    private final DisruptorAutobatcher<Long, TransactionSoftCacheStatus> autobatcher;
+
+    private final AtomicReference<PatchyCache> patchyCacheRef;
```

Comment:
You don't really need an atomic ref now that we are autobatching, volatile should be enough